torchprime/torch_xla_models/train.py

```diff
@@ -48,6 +49,31 @@
 assert xr.is_spmd() is True
 
 
+def _is_multi_slice() -> bool:
+  """
+  Check if we are running in multi-slice (or multi-pod) environment.
+  """
+  device_attributes = xr.global_runtime_device_attributes()
+  slice_indices = []
+  for d in device_attributes:
+    index = d.get("slice_index", "0")
+    assert isinstance(index, str), f"Expected slice_index to be a string, got {index}"
+    slice_indices.append(int(index))
+  num_slices = max(slice_indices) + 1
+  return num_slices > 1
+
+
+def _is_1d_sharding(mesh_values: tuple[int, ...]) -> bool:
```

Comment:
Would it be better if we moved these functions to a utils file and added tests for it. That protects us from any changes that happens to these torch_xla api

Reply:
Yes. I couldn't test `is_multi_slice` super extensively because we don't know what's the environment these tests will be run under, but at least I could add some tests and make sure they don't crash.

Reply:
Done